Automattic/wp-calypso
# Add site picker step to wooccommerce-install signup flow.

### Follow-up from https://github.com/Automattic/wp-calypso/pull/58441

~WIP,~ this needs a bunch of cleaning up, I've branched from another PR: https://github.com/Automattic/wp-calypso/pull/58333 that should be merged first. This PR is also waiting on the confirm step changes in https://github.com/Automattic/wp-calypso/pull/58024, I'll rework this PR to accommodate the new confirm steup once that is merged, and likely move over the install step in a same way as the transfer step done in 58333. This is needed to clean up the siteId dependency and reduce the props threading through the central woo install step (that I'd also like to drop at this point)

#### Changes proposed in this Pull Request

* Add site picker step to wooccommerce-install signup flow.

#### Testing instructions

* http://calypso.localhost:3000/start/woocommerce-install should redirect to the /select-site step and ask for a site selection.
* The confirm and transfer flow should still work as expected



![Screenshot 2021-11-23 at 14-39-35 Add WooCommerce to your site — WordPress com](https://user-images.githubusercontent.com/811776/142968101-1d7a3f2f-c2c2-494f-9706-ade3ec771546.png)


https://user-images.githubusercontent.com/77539/143307971-8542eca7-44d2-4ef8-bd81-edc1871cfc0e.mov



Related to https://github.com/Automattic/wp-calypso/issues/58229
